### PR TITLE
Remove unwanted jvmti error code for Java 8

### PR DESCRIPTION
**Pull request by keithc-ca**
`JVMTI_ERROR_UNSUPPORTED_REDEFINITION_CLASS_ATTRIBUTE_CHANGED` should only be declared for Java11+. There were comments to that effect which were removed by #9507: this omits the declaration in Java 8.

### CODE REVIEW

File: runtime/include/jvmti.h.m4 [hunk 1]
```diff
@@ -341,7 +341,10 @@ typedef enum jvmtiError {
 	JVMTI_ERROR_NAMES_DONT_MATCH = 69,
 	JVMTI_ERROR_UNSUPPORTED_REDEFINITION_CLASS_MODIFIERS_CHANGED = 70,
 	JVMTI_ERROR_UNSUPPORTED_REDEFINITION_METHOD_MODIFIERS_CHANGED = 71,
-	JVMTI_ERROR_UNSUPPORTED_REDEFINITION_CLASS_ATTRIBUTE_CHANGED = 72,
+dnl m4 can't understand (defined(J9VM_OPT_VALHALLA_NESTMATES))
```
Comment (DanHeidinga) [inline runtime/include/jvmti.h.m4:344]: This line can be removed.  Nestmates is an 11 feature so we don't need the specific ifdef for it anymore

Comment (keithc-ca) [reply runtime/include/jvmti.h.m4:344]: Perhaps it's time to remove all appearances of `J9VM_OPT_VALHALLA_NESTMATES` (in a separate PR)?

Comment (DanHeidinga) [reply runtime/include/jvmti.h.m4:344]: Agreed.  They should all become Java 11 version checks

Comment (keithc-ca) [reply runtime/include/jvmti.h.m4:344]: @DanHeidinga Do you also agree that change should be in a separate pull request or would you prefer I add those changes here?

Comment (DanHeidinga) [reply runtime/include/jvmti.h.m4:344]: My assumption, and sorry if it wasn't clear, is that we'd delete the lines in this PR and the larger cleanup of `J9VM_OPT_VALHALLA_NESTMATES` can be deferred to another issue

Comment (keithc-ca) [reply runtime/include/jvmti.h.m4]: Done.

**Comment by DanHeidinga** [comment]
Jenkins test sanity zlinux jdk8,jdk11

**Pull request merged**